runsc/cgroup/systemd.go

```diff
@@ -138,6 +139,48 @@ func (c *cgroupSystemd) Install(res *specs.LinuxResources) error {
 	return nil
 }
 
+// Update updates the cgroup resources of an existing systemd unit.
+func (c *cgroupSystemd) Update(res *specs.LinuxResources) error {
+	log.Debugf("Updating systemd cgroup resource controller under %v", c.Parent)
+	for controllerName, ctrlr := range controllers2 {
+		// First check if our controller is found in the system.
+		found := false
+		for _, knownController := range c.Controllers {
+			if controllerName == knownController {
+				found = true
+				break
+			}
+		}
+		if found {
+			props, err := ctrlr.generateProperties(res)
+			if err != nil {
+				return err
+			}
+			c.properties = append(c.properties, props...)
+			continue
+		}
+		if ctrlr.optional() {
+			if err := ctrlr.skip(res); err != nil {
+				return err
+			}
+		} else {
+			return fmt.Errorf("mandatory cgroup controller %q is missing for %q", controllerName, c.Path)
+		}
+	}
```

Comment:
Refactor this into a new function, maybe `update()` and replace line 114-138 with call to `c.update(res)`.

Reply:
sounds good, i think let's name it `updateControllerProps()`, to make it similar to `addProp()` and `addIOProps()` which already exists in `cgroupSystemd`.